libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -0,0 +1,1000 @@
+      (EOSIO_CDT_CAT(EOSIO_CDT_KV_INDEX_TEST_,                                                                         \
+           EOSIO_CDT_EXPAND(EOSIO_CDT_KV_INDEX_TEST EOSIO_CDT_KV_INDEX ## index_name ()))))(index_name)
+
+#define EOSIO_CDT_KV_INDEX_CONSTRUCT(value_class, index_name)                                                          \
+   EOSIO_CDT_APPLY(EOSIO_CDT_KV_FIX_INDEX_CONSTRUCT,                                                                   \
+      (EOSIO_CDT_CAT(EOSIO_CDT_KV_INDEX_TEST_,                                                                         \
+           EOSIO_CDT_EXPAND(EOSIO_CDT_KV_INDEX_TEST EOSIO_CDT_KV_INDEX ## index_name ()))))(value_class, index_name)
+
+#define EOSIO_CDT_CREATE_KV_INDEX(r, value_class, i, index_name)                                                       \
+   EOSIO_CDT_KV_INDEX_TYPE(index_name)<EOSIO_CDT_GET_RETURN_T(value_class, index_name)> EOSIO_CDT_KV_INDEX_NAME(index_name, i) EOSIO_CDT_KV_INDEX_CONSTRUCT(value_class, index_name);
+
+#define EOSIO_CDT_CREATE_KV_INDICES(value_class, ...)                                                                  \
+   BOOST_PP_SEQ_FOR_EACH_I(EOSIO_CDT_CREATE_KV_INDEX, value_class, BOOST_PP_VARIADIC_TO_SEQ(__VA_ARGS__))
+
+#define EOSIO_CDT_LIST_KV_INDEX(r, data, index_name)                                                                   \
+   ,&index_name
+
+#define EOSIO_CDT_LIST_KV_INDICES(...)                                                                                 \
+   BOOST_PP_SEQ_FOR_EACH(EOSIO_CDT_LIST_KV_INDEX, _, BOOST_PP_VARIADIC_TO_SEQ(__VA_ARGS__))
+
+/**
+ * @brief Macro to define a table.
+ * @details The resulting table will have member fields on it that match 1-1 with the names of the
+ * fields passed into the list. See example for further clarification.
+ *
+ * @param table_class - The name of the class of the user defined table that inherits from eosio::kv_table
+ * @param value_class - The name of the class of the data stored as the value of the table
+ * @param table_name  - The name of the table
+ * @param db_name     - The type of the EOSIO Key Value database. Defaulted to eosio.kvram
+ * @param ...         - A variadic list of 1 or more indexes to define on the table.
+ */
+#define DEFINE_TABLE(table_class, value_class, table_name, db_name, /*indices*/...)                                    \
+   struct table_class : eosio::kv_table<value_class> {                                                                 \
+      EOSIO_CDT_CREATE_KV_INDICES(value_class, __VA_ARGS__)                                                            \
+                                                                                                                       \
+      table_class(eosio::name contract_name) {                                                                         \
+         init(contract_name, table_name##_n, db_name EOSIO_CDT_LIST_KV_INDICES(__VA_ARGS__));                          \
```

Comment:
If you make this
```cpp
init(contract_name, eosio::name(table_name), eosio::name(db_name),  EOSIO_CDT_LIST_KV_INDICES(__VA_ARGS__));   
```
then the user of the macro has the flexibility of providing either a regular string (and not just as a string literal but also as the name of some `const` global variable if desired) or a `""_n` string and it would work either way.